microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/ls/properties/PropertiesFileTextDocumentService.java

```diff
@@ -176,60 +184,61 @@ public CompletableFuture<List<Either<SymbolInformation, DocumentSymbol>>> docume
 	@Override
 	public CompletableFuture<Either<List<? extends Location>, List<? extends LocationLink>>> definition(
 			DefinitionParams params) {
-		MicroProfileProjectInfoParams projectInfoParams = createProjectInfoParams(params.getTextDocument());
-		return getProjectInfoCache().getProjectInfo(projectInfoParams).thenComposeAsync(projectInfo -> {
-			if (projectInfo.getProperties().isEmpty()) {
-				return CompletableFuture.completedFuture(null);
+		return getPropertiesModel(params.getTextDocument(), (document, cancelChecker) -> {
+			MicroProfileProjectInfoParams projectInfoParams = createProjectInfoParams(params.getTextDocument());
+			MicroProfileProjectInfo projectInfo = getProjectInfoCache().getProjectInfo(projectInfoParams).getNow(null);
+			if (projectInfo == null || projectInfo.getProperties().isEmpty()) {
+				return null;
 			}
-			// then get the Properties model document
-			return getDocument(params.getTextDocument().getUri()).getModel().thenComposeAsync(document -> {
-				return getPropertiesFileLanguageService().findDefinition(document, params.getPosition(), projectInfo,
-						microprofileLanguageServer.getLanguageClient(), isDefinitionLinkSupport());
-			});
+			return Tuple.two(Tuple.two(document, projectInfo), cancelChecker);
```

Comment:
Why you canot just return 
getPropertiesFileLanguageService().findDefinition here?

Reply:
I thought I would be able to do this as well, but since `findDefinition` returns a `CompletableFuture` (since it makes a call to `microprofile/propertyDefinition` on the JDT side), there is a compilation error if I try to combine it that way.

Reply:
Ok thanks for your explanation.